context/app/static/js/pages/Genes/Genes.jsx

```diff
@@ -5,12 +5,14 @@ import TableOfContents from 'js/shared-styles/sections/TableOfContents';
 import { getSections } from 'js/shared-styles/sections/TableOfContents/utils';
 import Summary from 'js/components/genes/Summary';
 import { FlexRow, Content } from './style';
-import { useGeneCommonName } from './hooks';
+import { useGeneCommonName, useGeneData, useGeneId } from './hooks';
 
 const summaryId = 'Summary';
 
 function Genes({ geneSymbol }) {
   const { geneCommonName, isLoading } = useGeneCommonName(geneSymbol);
+  const geneSummary = useGeneData(geneSymbol);
+  const geneId = useGeneId(geneSymbol);
```

Comment:
- Since these hooks' values are just getting passed down to `SummaryBody`, we could pass down `geneSymbol` instead and call these hooks from `SummaryBody`; being able to use hooks colocated with where the data is used without having to worry about duplicate requests is one of the benefits of using SWR and makes future troubleshooting easier!
  - I think it's also worth considering creating a context to hold the `geneSymbol` value so you don't have to [prop drill](https://www.geeksforgeeks.org/what-is-prop-drilling-and-how-to-avoid-it/) the `geneSymbol` down to every component on the page; since the `geneSymbol` should change after page routing, so keeping it in Context shouldn't cause any performance concerns and cleans up some visual noise in components where the symbol is not used, e.g. the `<Summary>` component.
- To support that revision (moving the hooks down to the `SummaryBody`), the `isLoading` check on line 26 could be revised so that it only blocks the `geneCommonName` display in the title.

Lines 26-28 would be replaced with:
```
const title = `${geneSymbol.toUpperCase()}} ${ isLoading ? `(${geneCommonName})` : null }`
```

Then the `title` would be used in line 38:
```
<Typography variant="h1" component="h1">
  {title}
</Typography>
```

This way, the fetch of the `geneCommonName` shouldn't block any fetch operations that don't depend on the name being present, which should make the page load faster for users (with requests firing in parallel instead of serially)

Reply:
As mentioned before, I was working on the wrong branch! I reverted my changes and pushed. I will make sure to still apply your comments on the branch.

Reply:
You shouldn't need context with SWR. SWR caches results, right?

Reply:
Yup, no need for an extra context with SWR after v1, it has a global cache by default now.

I’m referring to moving the “geneSymbol” for the page into its own context, not the products of the fetches that use it.